substrate/client/statement-store/src/lib.rs

```diff
@@ -779,6 +786,22 @@ impl StatementStore for Store {
 		Ok(result)
 	}
 
+	fn take_recent_statements(&self) -> Result<Vec<(Hash, Statement)>> {
+		let mut index = self.index.write();
+		let recent = index.take_recent();
+		let mut result = Vec::with_capacity(recent.len());
+		for h in recent {
+			let encoded = self.db.get(col::STATEMENTS, &h).map_err(|e| Error::Db(e.to_string()))?;
+			if let Some(encoded) = encoded {
+				if let Ok(statement) = Statement::decode(&mut encoded.as_slice()) {
+					let hash = statement.hash();
+					result.push((hash, statement));
```

Comment:
```suggestion
					result.push((h, statement));
```

I assume `h` is the hash already?